packages/muelu/src/Misc/MueLu_InverseApproximationFactory_def.hpp

```diff
@@ -143,6 +160,60 @@ namespace MueLu {
     Set(currentLevel, "Ainv", Ainv);
   }
 
+  template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+  RCP<const Xpetra::CrsGraph<LocalOrdinal, GlobalOrdinal, Node>>
+  InverseApproximationFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::GetStaticPattern(const RCP<const CrsGraph>& sparsityPattern, int power) const {
+
+    // gather missing rows from other procs to generate an overlapping map
+    const RCP<const Import> rowImport = ImportFactory::Build(sparsityPattern->getRowMap(), sparsityPattern->getColMap());
```

Comment:
Does sparsity pattern maybe already have an importer?

Reply:
I don't think so, at least I don't see were one has been initialized.

Reply:
`sparsityPattern` is the graph of A, right? So you could check that `sparsityPattern->getImporter()` returns something non-null.

Reply:
Yep, I checked, it returns null :sweat_smile: .

Reply:
The `Xpetra::CrsGraph()` can return an importer via `getImporter()`. However, this is only implemented for the `Xpetra::TpetraCrsGraph`.

The `Xpetra::EpetraCrsGraph()` returns a hard-coded `Teuchos::null`. When looking at the implementation of the `Epetra_CrsGraph` in `Epetra`, it looks like the graph is capable of returning an importer. So, it might be just the incompleteness of `Xpetra` that comes into play here. (@maxfirmbach's, if you have tested this in your application code, then you have tested the `Epetra`- route through `Xpetra`.)

Reply:
@mayrmt No, it's implemented for Epetra as well. The two specializations for EpetraNode return the importer.